JOSESwift/Sources/JWE.swift

```diff
@@ -185,7 +187,9 @@ public struct JWE {
         }
 
         do {
-            return Payload(try decrypter.decrypt(context))
+            let compressor = try CompressorFactory.makeCompressor(algorithm: header.compressionAlgorithm)
+            let decryptedData = try decrypter.decrypt(context)
+            return Payload(try compressor.decompress(data: decryptedData))
         } catch {
```

Comment:
I think we could explicitly catch the compression error and propagate it - what do you think? Same for decryption.

Compression is before/after the encryption/decryption process, so security wise we are fine with throwing the specific error. And it gives the user correct insight into what actually happened. The tests would be more obvious too.